qcodes/instrument_drivers/signal_hound/USB_SA124B.py

```diff
@@ -1,226 +1,447 @@
+        err = self.dll.saConfigSweepCoupling(
+            self.deviceHandle, ct.c_double(self.rbw()),
+            ct.c_double(self.vbw()), reject_var)
+        self.check_for_error(err, 'saConfigSweepCoupling')
+
+
+        modeOpts = {
+            'sweeping': self.hf.sa_SWEEPING,
+            'real_time': self.hf.sa_REAL_TIME,  # not implemented
+            'IQ': self.hf.sa_IQ,  # not implemented
+            'idle': self.hf.sa_IDLE
+        }
+        mode = modeOpts[self.device_mode()]
+        # the third argument to saInitiate is a flag that is
+        # currently not used
+        err = self.dll.saInitiate(self.deviceHandle, mode, 0)
+        if err == saStatus.saInvalidParameterErr:
+            extrainfo = """
+                 In real-time mode, this value may be returned if the span
+                 limits defined in the API header are broken. Also in
+                 real-time mode, this error will be returned if the
+                 resolution bandwidth is outside the limits defined in
+                 the API header.
+                 In time-gate analysis mode this error will be returned if
+                 span limits defined in the API header are broken. Also in
+                 time gate analysis, this error is returned if the
+                 bandwidth provided require more samples for processing
+                 than is allowed in the gate length. To fix this
+                 increase rbw/vbw.
+             """
+        elif err == saStatus.saBandwidthErr:
+            extrainfo = 'RBW is larger than your span. (Sweep Mode)!'
+        else:
+            extrainfo = None
+        self.check_for_error(err, 'saInitiate', extrainfo)
+
+        self._parameters_synced = True
 
-    @classmethod
-    def default_server_name(cls, **kwargs):
-        return 'USB'
+    def configure(self) -> None:
+        """
+        Syncs parameters to the Instrument and updates the setpoint of the
+        trace.
+        """
+        self.sync_parameters()
+        self._update_trace()
 
-    def openDevice(self):
+    def openDevice(self) -> None:
+        """
+        Opens connection to the instrument
+        """
         log.info('Opening Device')
         self.deviceHandle = ct.c_int(0)
         deviceHandlePnt = ct.pointer(self.deviceHandle)
-        ret = self.dll.saOpenDevice(deviceHandlePnt)
-        if ret != self.saStatus['saNoError']:
-            if ret == self.saStatus['saNullPtrErr']:
-                raise ValueError('Could not open device due to '
-                                 'null-pointer error!')
-            elif ret == self.saStatus['saDeviceNotOpenErr']:
-                raise ValueError('Could not open device!')
-            else:
-                raise ValueError('Could not open device due to unknown '
-                                 'reason! Error = %d' % ret)
-
+        err = self.dll.saOpenDevice(deviceHandlePnt)
+        self.check_for_error(err, 'saOpenDevice')
         self.devOpen = True
```

Comment:
Perhaps this should just go. We dont have anything like this normally. The driver knows it self if the instrument is closed